packages/components/src/suggestions/item.jsx

```diff
@@ -26,12 +27,14 @@ class Item extends PureComponent {
 
 	/**
 	 * Highlights the part of the text that matches the query.
+	 *
 	 * @param  {string} text  Text.
 	 * @param  {string} query The text to be matched.
-	 * @returns {element}      A React element including the highlighted text.
+	 *
+	 * @returns {Array< ReactElement< JSX.IntrinsicElements[ 'span' ] > >} An element including the highlighted text.
 	 */
 	createTextWithHighlight( text, query ) {
-		const re = new RegExp( '(' + query + ')', 'gi' );
+		const re = new RegExp( '(' + escapeRegexp( query ) + ')', 'gi' );
```

Comment:
We used [lodash.escapeRegExp](https://github.com/lodash/lodash/blob/4.1.2-npm-packages/lodash.escaperegexp/index.js) [client/components/suggestion-search/index.jsx](https://github.com/Automattic/wp-calypso/blob/master/client/components/suggestion-search/index.jsx#L147), but this one looks [much smaller](https://github.com/sindresorhus/escape-string-regexp/blob/master/index.js). Nice!

Maybe even small enough to add to our own utils lib instead of pulling it from npm?

Reply:
Thanks for calling this out, I'm going to revisit this.

Reply:
I've removed that outer escaping, which should be redundant with this PR. Can you test to make sure there are no regressions?

Reply:
Exploring replacement across the board: #38839